native/cocos/platform/android/java/src/com/cocos/lib/CocosEngine.java

```diff
@@ -0,0 +1,179 @@
+/****************************************************************************
+ Copyright (c) 2023 Xiamen Yaji Software Co., Ltd.
+
+ http://www.cocos.com
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated engine source code (the "Software"), a limited,
+ worldwide, royalty-free, non-assignable, revocable and non-exclusive license
+ to use Cocos Creator solely to develop games on your target platforms. You shall
+ not use Cocos Creator software for developing other software or tools that's
+ used for developing games. You are not granted to publish, distribute,
+ sublicense, and/or sell copies of Cocos Creator.
+
+ The software or tools in this License Agreement are licensed, not sold.
+ Xiamen Yaji Software Co., Ltd. reserves all rights not expressly granted to you.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ ****************************************************************************/
+
+package com.cocos.lib;
+
+import android.app.Activity;
+import android.content.Context;
+import android.media.AudioManager;
+import android.os.Handler;
+import android.view.SurfaceView;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.FrameLayout;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.List;
+
+
+class CocosEngine {
+    private static WeakReference<CocosEngine> mRefCocosEngine;
+
+    private CocosAudio mAudio;
+    private Context mContext;
+
+    private CocosSensorHandler mSensorHandler;
+
+    private CocosWebViewHelper mWebViewHelper = null;
+    private CocosVideoHelper mVideoHelper = null;
+    private FrameLayout mRootLayout;
+    private final Handler mHandler;
+
+    private List<CocosSurfaceView> mSurfaceViewArray;
+    private SurfaceView mSurfaceView;
+
+    private native void initEnvNative(Context context);
+
+    CocosEngine(Context context, String libName) {
+        mRefCocosEngine = new WeakReference<>(this);
+        mContext = context;
+        mAudio = new CocosAudio(context);
+        mHandler = new Handler(context.getMainLooper());
+        System.loadLibrary(libName);
+        initEnvNative(context);
+    }
+
+    void destroy() {
+        mRefCocosEngine.clear();
+        CocosHelper.unregisterBatteryLevelReceiver(mContext);
+        mAudio.unregisterAudioFocusListener(mContext);
+        CanvasRenderingContext2DImpl.destroy();
+        GlobalObject.destroy();
+        mAudio = null;
+        mContext = null;
+    }
+
+    void init() {
```

Comment:
When will init be invoked?

Reply:
it's a module of CocosEngine

Reply:
It can be init at construct, but Actvity should be called after super.oncreate

Reply:
and call initWithView instead